src/wallet/wallet.cpp

```diff
@@ -3826,14 +3826,19 @@ std::shared_ptr<CWallet> CWallet::CreateWalletFromFile(interfaces::Chain& chain,
         }
     }
 
-    if (!gArgs.GetArg("-addresstype", "").empty() && !ParseOutputType(gArgs.GetArg("-addresstype", ""), walletInstance->m_default_address_type)) {
-        error = strprintf(_("Unknown address type '%s'"), gArgs.GetArg("-addresstype", ""));
-        return nullptr;
+    if (!gArgs.GetArg("-addresstype", "").empty()) {
+        if (!ParseOutputType(gArgs.GetArg("-addresstype", ""), walletInstance->m_default_address_type)) {
+            error = strprintf(_("Unknown address type '%s'"), gArgs.GetArg("-addresstype", ""));
+            return nullptr;
+        }
     }
 
-    if (!gArgs.GetArg("-changetype", "").empty() && !ParseOutputType(gArgs.GetArg("-changetype", ""), walletInstance->m_default_change_type)) {
-        error = strprintf(_("Unknown change type '%s'"), gArgs.GetArg("-changetype", ""));
-        return nullptr;
+    if (!gArgs.GetArg("-changetype", "").empty()) {
+        walletInstance->m_default_change_type.emplace();
+        if (!ParseOutputType(gArgs.GetArg("-changetype", ""), *walletInstance->m_default_change_type)) {
```

Comment:
fa078530bc3824d7d5a1cc4f39b617054b08d77e

nit, I'd rather ditch the above, declare a `OutputType change_type` in this scope, and assign to `walletInstance->m_default_change_type` if parse succeeds. Not really relevant but this approach changes `m_default_change_type` for no good reason if parse fails.

Reply:
It doesn't matter here, but I changed it as a belt-and-suspenders measure